### ISSUE TITLE
[New package]: metals

### ISSUE THREAD

**Issue by yungParrot** [williamboman/mason.nvim#369]
Package name
metals

Package homepage
https://scalameta.org/metals/

Languages
scala

How is this package distributed?
As `.zip` or `.tar.gz.` files. [See releases](https://github.com/scalameta/metals/releases)

**Comment by cjgratacos** [comment]
This one will be great to have.

**Comment by benjstephenson** [comment]
I was taking a look at this today (not done much Lua in the past so a bit slow going for me), I was wondering if it makes sense to use the metals package directly or via the [nvim-metals](https://github.com/scalameta/nvim-metals) package as it does a lot of config already that would probably end up being duplicated?

**Comment by williamboman** [comment]
Yeah the nvim-metals plugin provides a `:MetalsInstall` command. I believe it's only available via Coursier, for which support would need to be added from scratch in Mason (similar to [npm](https://github.com/williamboman/mason.nvim/blob/7ff829a329483a5dc4f86c7b9d1e5346e7a715da/lua/mason-core/managers/npm/init.lua) for example). There might be some aspects of Coursier that makes it a bad fit for Mason - for example inability to customize installations (we don't want to be installing packages outside of Mason's install dir for instance).

For people who want to use metals I'd strongly recommend using the `nvim-metals` plugin (and managing ones metals installation through it)

**Comment by ckipp01** [comment]
Hey! 👋🏼 the author of `nvim-metals` here and just wanted to jump in here since I had a user asking about this on our Discord.

> I believe it's only available via Coursier, for which support would need to be added from scratch in Mason

Correct, we actually don't provide executable binaries by default. The zip files that someone linked about in the releases are just the sources. Most Scala users will hopefully be familiar enough with Coursier to use it to install since it's used to even install Scala itself. `nvim-metals` actually doesn't use the defualt `cs install`, but rather a more complex `bootstrap` command that gives us the ability to do thing like easily tell it _where_ to generate the binary, plus some other goodies.

So yea, I sort of agree that if a user is looking for an _easy_ way to get metals installed, just let `nvim-metals` do it for you. There is also the option of having it globally installed and having `nvim-metals` use that.

_If_ mason really wanted to support it, one thing that we could maybe do it have mason piggy back on `nvim-metals` and just use the Lua api we provide to install Metals. Implementing coursier in mason may be overkill since so few apps will be installed that way. Either way, if there is anything I can do on my end to help the situation, don't hesitate to ping me.

**Comment by ckipp01** [comment]
Hey @mvillafuertem I'm not actively working or involved in this really. My own opinion as the author of nvim-metals is to either let nvim-metals handle the install, or use coursier directly to install Metals and then tell nvim-metals to use that global installation. Seeing this message from @williamboman 

> For people who want to use metals I'd strongly recommend using the nvim-metals plugin (and managing ones metals installation through it)

My guess is that we're sort of aligned here.
